src/Raven.Client/Util/JavascriptConversionExtensions.cs

```diff
@@ -125,6 +126,22 @@ public override void ConvertToJavascript(JavascriptConversionContext context)
                         context.PreventDefault();
                         context.Visitor.Visit(methodCallExpression.Arguments[0]);
                         return;
+                    case "FirstOrDefault":
+                    case "First":
+                        context.PreventDefault();
+                        context.Visitor.Visit(methodCallExpression.Arguments[0]);
+                        var writer = context.GetWriter();
+                        using (writer.Operation(methodCallExpression))
+                        {
+                            if (methodCallExpression.Arguments.Count > 1)
+                            {
+                                writer.Write(".filter");
```

Comment:
Use `find` instead of `filter`

Reply:
I tried using find, it seems like it's not supported in jint, that's why i went with filter..

running this query in studio :
from Regions as r
select {  T : r.Territories.find(function(x){return x !== null;}) }

gives this error :
Raven.Client.Exceptions.Documents.Patching.JavaScriptException: At 4:2
{"name":"TypeError","message":"Object has no method 'find'","callstack":" at find(Esprima.Ast.FunctionExpression) @  8:3\r\n at find(Esprima.Ast.FunctionExpression) @  8:3\r\n"} ---> Jint.Runtime.JavaScriptException: Object has no method 'find'
   at Jint.Native.Function.ScriptFunctionInstance.Call(JsValue thisArg, JsValue[] arguments)
   at Raven.Server.Documents.Patch.ScriptRunner.SingleRun.Run(DocumentsOperationContext ctx, String method, Object[] args) in C:\Builds\RavenDB-4.0-Nightly\src\Raven.Server\Documents\Patch\ScriptRunner.cs:line 426